cov.el

```diff
@@ -260,6 +260,26 @@ that filename. Otherwise search for the first matching pattern in
                                 cov-lcov-patterns)))
     (when lcov-file (cons lcov-file 'lcov))))
 
+(defun cov--locate-gcov-cmake (file-dir file-name)
+  "Locate a gcov coverage file from FILE-DIR for FILE-NAME.
+This works with CMake-based projects, by constructing the path to the `.gcov'
+file from object file's path extracted from the \"compile_commands.json\"."
+  (when-let* ((root (project-root (project-current)))
+              (compile-commands (expand-file-name "compile_commands.json" root)))
+    (when (file-exists-p compile-commands)
+      (when-let* ((file-cmd (cl-find-if
+                             (lambda (entry)
+                               (equal (file-truename (alist-get 'file entry))
+                                      (expand-file-name file-name file-dir)))
+                             (json-read-file compile-commands)))
+                  (command (alist-get 'command file-cmd))
+                  (directory (alist-get 'directory file-cmd)))
+        (let* ((obj-file (when (string-match
+                                (concat "-o \\(?1:.*" (regexp-quote (concat file-name ".o")) "\\)")
```

Comment:
This match looks fragile to me. What if there is no space between -o and the file name for instance?

Reply:
Well spotted, I've changed it to " -o *"..., while accepting matches "file.cpp.o" and "file.o" for a "file.cpp". This is more generic than assuming object files will have the "file.cpp.o" format (which is correct for CMake, but not necessary valid for other build tools).